### PR TITLE
Switch to sbt-typelevel

### PR DESCRIPTION
**Pull request by mpilquist**

### CODE REVIEW

**Comment by mpilquist** [comment]
@armanbilge Can you do a quick review on this?

File: build.sbt [hunk 1]
```diff
@@ -1,19 +1,14 @@
...
 ThisBuild / organizationName := "Comcast Cable Communications Management, LLC"
 
-ThisBuild / homepage := Some(url("https://github.com/comcast/ip4s"))
 ThisBuild / startYear := Some(2018)
 
-ThisBuild / publishGithubUser := "mpilquist"
-ThisBuild / publishFullName := "Michael Pilquist"
-
 ThisBuild / developers ++= List(
+  Developer("mpilquist", "Michael Pilquist", "@mpilquist", url("https://github.com/mpilquist")),
```
Comment (armanbilge) [inline build.sbt:11]: These can be shortened.
```suggestion
  tlGitHubDev("mpilquist", "Michael Pilquist"),
```

File: build.sbt [hunk 2]
```diff
@@ -22,38 +17,22 @@ ThisBuild / githubWorkflowJavaVersions := Seq(JavaSpec.temurin("8"))
...
 
-ThisBuild / scmInfo := Some(
-  ScmInfo(
-    url("https://github.com/comcast/ip4s"),
-    "git@github.com:comcast/ip4s.git"
-  )
-)
-
-ThisBuild / testFrameworks += new TestFramework("munit.Framework")
+ThisBuild / tlCiReleaseBranches := List("main")
```
Comment (armanbilge) [inline build.sbt:22]: This can actually be omitted, it's already the default.

File: build.sbt [hunk 3]
```diff
@@ -22,38 +17,22 @@ ThisBuild / githubWorkflowJavaVersions := Seq(JavaSpec.temurin("8"))
...
 
 ThisBuild / doctestTestFramework := DoctestTestFramework.ScalaCheck
 
 ThisBuild / scalafmtOnCompile := true
```
Comment (armanbilge) [inline build.sbt:26]: 😡

File: build.sbt [hunk 4]
```diff
@@ -86,21 +65,10 @@ lazy val core = crossProject(JVMPlatform, JSPlatform)
   .settings(
     name := "ip4s-core",
     libraryDependencies ++= {
-      if (isDotty.value) Nil else List("org.scala-lang" % "scala-reflect" % scalaVersion.value % "provided")
+      if (scalaVersion.value.startsWith("3")) Nil
```
Comment (armanbilge) [inline build.sbt:68]:
```suggestion
      if (tlIsScala3.value) Nil
```

File: build.sbt [hunk 5]
```diff
@@ -86,21 +65,10 @@ lazy val core = crossProject(JVMPlatform, JSPlatform)
...
-    Test / scalafmt / unmanagedSources := (Test / scalafmt / unmanagedSources).value.filterNot(
-      _.toString.endsWith("Literals.scala")
-    ),
-    Compile / unmanagedSourceDirectories ++= {
-      val major = if (isDotty.value) "-3" else "-2"
-      List(CrossType.Pure, CrossType.Full).flatMap(
-        _.sharedSrcDir(baseDirectory.value, "main").toList.map(f => file(f.getPath + major))
-      )
-    }
+    scalacOptions := scalacOptions.value.filterNot(_ == "-source:3.0-migration")
```
Comment (armanbilge) [inline build.sbt:71]: Just for my learning, why does this option not work here?

**Review by armanbilge — COMMENTED** [review]
Looks good! Some additional shortcuts available.

File: build.sbt (hunk 5 above)
Comment (mpilquist) [reply build.sbt:71]: It disables certain Scala 3 only features - in this case, the macro support.

File: build.sbt [hunk 6]
```diff
@@ -1,19 +1,14 @@
...
-ThisBuild / homepage := Some(url("https://github.com/comcast/ip4s"))
 ThisBuild / startYear := Some(2018)
 
-ThisBuild / publishGithubUser := "mpilquist"
-ThisBuild / publishFullName := "Michael Pilquist"
-
 ThisBuild / developers ++= List(
+  tlGitHubDev("mpilquist", "Michael Pilquist"),
   Developer("matthughes", "Matt Hughes", "@matthughes", url("https://github.com/matthughes")),
   Developer("nequissimus", "Tim Steinbach", "@nequissimus", url("https://github.com/nequissimus"))
```
Comment (armanbilge) [inline build.sbt:13]:
```suggestion
  tlGitHubDev("matthughes", "Matt Hughes"),
  tlGitHubDev("nequissimus", "Tim Steinbach")
```

**Pull request merged**